neo4j/config.go

```diff
@@ -40,11 +40,19 @@ type Config struct {
 
 	// Logging target the driver will send its log outputs
 	//
-	// Possible to use custom logger (implement neo4j.log.Logger interface) or
-	// use neo4j.ConsoleLog.
+	// Possible to use custom logger (implement log.Logger interface) or
+	// use neo4j.ConsoleLogger.
 	//
-	// default: No Op Logger (neo4j.VoidLog)
+	// default: No Op Logger (log.Void)
 	Log log.Logger
+
+	// Logging target the driver will send its Bolt message traces
+	//
+	// Possible to use custom logger (implement log.BoltLogger interface) or
+	// use neo4j.ConsoleBoltLogger.
+	//
+	// default: No Op Logger (log.Void)
```

Comment:
If you want almost zero cost logging you could make it a pointer and have conditionals on bolt level to log. As you saw from the flood of messages that code runs quite a lot so any function call that can be avoided will save cycles. You probably need some trickery to handle nil since you use interfaces. A simple way is to have the external interface wrapped in an outer internal struct.

type interface BoltLogger {
   Message
}

type InternalBoltLogger struct {
     logger BoltLogger
}

An instance of above is what bolt uses and might be nil or an instance. Another variant to nil check is to add a variable in the struct that indicates if it is enabled or not. (if you have an interface you would need a func call for the check)

Reply:
Now done